### PR TITLE
XSUP-35945 - handle 429 rate limit errors

### PR DESCRIPTION
**Pull request by RotemAmit**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: [link to the issue](https://jira-dc.paloaltonetworks.com/browse/XSUP-35945)

## Description
According to crowdstrike, 429 is received from requests with missing, malformed, or expired bearer tokens.
To better handle 429 rate limit error in crowdstrike falcon integration, when a 429 is received from the HTTP request, we will generate a new token, and try again.

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

**Review by MLainer1 — COMMENTED** [review]
👍

File: Packs/CrowdStrikeFalcon/Integrations/CrowdStrikeFalcon/CrowdStrikeFalcon.py [hunk 1]
```diff
@@ -340,21 +406,21 @@ def http_request(method, url_suffix, params=None, data=None, files=None, headers
...
-            params=params,
-            data=data,
+        res = generic_http_request(
+            method=method,
+            server_url=SERVER,
             headers=headers,
+            url_suffix=url_suffix,
+            data=data,
             files=files,
-            json=json,
```
Comment (eyalpalo) [inline Packs/CrowdStrikeFalcon/Integrations/CrowdStrikeFalcon/CrowdStrikeFalcon.py:356]: what about the json? I can see it is removed

**Review by eyalpalo — APPROVED** [review]

**Comment by RotemAmit** [comment]
The test playbook fails due to a known [issue](https://jira-dc.paloaltonetworks.com/browse/CIAC-10232). @dantavori Can you force merge the PR?

**Comment by RotemAmit** [comment]
The test playbook is failing due to a [known issue](https://jira-dc.paloaltonetworks.com/browse/CIAC-10232). @dantavori can you force merge the PR?

**Pull request merged**